### PR TITLE
[SB][106947438] Clear markers (cluster agnostic)

### PR DESCRIPTION
**Pull request by sbrauer**
The code used to clear markers via the clusterer.
Now that clustering is optional (via shouldCluster)
we can't depend on a clusterer to do the clearing.

### CODE REVIEW

File: app/coffeescript/components/ui/markers.coffee [hunk 1]
```diff
@@ -64,6 +62,16 @@ define [
       @updateListingsCount()
       @trigger 'uiSetMarkerInfoWindow'
 
+    @clearAllMarkers = ->
```
Comment (c0) [inline app/coffeescript/components/ui/markers.coffee:65]: What do you think about moving `markerClusterer.unbindMarkers` here? That way it's applied w/o the clusterer.

Comment (sbrauer) [inline app/coffeescript/components/ui/markers.coffee:65]: Not sure I understand what you're suggesting. Would you show me the code you have in mind?

Comment (c0) [inline app/coffeescript/components/ui/markers.coffee:65]: https://github.com/rentpath/map.js/blob/dev/app/coffeescript/components/mixins/clusters.coffee#L26-L28

**Comment by sbrauer** [comment]
@c0 - I think I did what you were suggesting. Lemme know.

It's weird, I don't see anything that ever called `unbindMarkers` other than `clearMarkers`, and I don't see anything that calls `clearMarkers`.
Even if it did, there'd be nothing to iterate over when calling `for marker in @attr.markerClusterer` since coffee compiles that to a for loop over `markerClusterer.length`, which is undefined.
Almost tempted to remove `clearMarkers` and `unbindMarkers` from clusters.coffee.

**Comment by c0** [comment]
@sbrauer `clearMarkers` is called by `ui/markers.coffee` (before this PR): https://github.com/rentpath/map.js/blob/dev/app/coffeescript/components/ui/markers.coffee#L51

It makes more sense that it's all in `ui/markers` rather than `mixins/clusters`.

**Comment by sbrauer** [comment]
I think that's actually calling the clearMarkers method of the 3rd party library (http://google-maps-utility-library-v3.googlecode.com/svn/trunk/markerclustererplus/docs/reference.html).

**Comment by c0** [comment]
Ah, you're right. That's confusing.

**Comment by c0** [comment]
:+1: Please resolve merge conflicts.

**Comment by sbrauer** [comment]
Conflict (which was in one of the compiled JS files) resolved.

**Comment by c0** [comment]
:+1:

**Pull request merged**